test/unit/licence_location_identifier_test.rb

```diff
@@ -2,93 +2,109 @@
+      "name" => "South Ribble Borough Council",
+      "type" => "DIS",
+      "codes" => {
+        "govuk_slug" => "south-ribble-borough-council",
+      },
+    }
+
+    @leyland_county_ward = {
+      "name" => "Leyland South West",
+      "type" => "CED",
+    }
+
+    @south_ribble_parliamentary_constituency = {
+      "name" => "South Ribble",
+      "type" => "WMC",
+    }
 
-    should "select the correct tier authority from geostack providing a district and county" do
-      geostack = {
-        "council" => [
-          {"id" => 1, "name" => "Lancashire County Council", "type" => "CTY", "govuk_slug" => "lancashire-county-council"},
-          {"id" => 2, "name" => "South Ribble Borough Council", "type" => "DIS", "govuk_slug" => "south-ribble-borough-council"},
-          {"id" => 3, "name" => "Leyland South West", "type" => "CED" },
-          {"id" => 4, "name" => "South Ribble", "type" => "WMC" }
-        ]
-      }
-      slug = LicenceLocationIdentifier.find_slug(geostack, @artefact)
+    @shropshire_unitary_authority = {
+      "name" => "Shropshire Council",
+      "type" => "UTA",
+      "codes" => {
+        "govuk_slug" => "shropshire-council",
+      },
+    }
+
+    @shrewsbury_civil_parish = {
+      "name" => "Shrewsbury_civil_parish",
+      "type" => "CPC",
+    }
+  end
+
+  # SOMETHING WRONG HERE, ONLY ONE ARTEFACT BEING CREATED
+  context "given an artefact exists with a local service for the district/unitary tiers" do
+    should "select the correct tier authority from areas providing a district and county" do
+      areas = [
+        @lancashire_county_council,
+        @south_ribble_borough_council,
+        @leyland_county_ward,
+        @south_ribble_parliamentary_constituency,
+      ]
+      slug = LicenceLocationIdentifier.find_slug(areas, @artefact)
 
       assert_equal "lancashire-county-council", slug
     end
 
     should "return nil if no authorities match" do
-      geostack = {
-        "council" => [
-          {"id" => 3, "name" => "Leyland South West", "type" => "CED" },
-          {"id" => 4, "name" => "South Ribble", "type" => "WMC" }
-        ]
-      }
-      slug = LicenceLocationIdentifier.find_slug(geostack, @artefact)
+      areas = [@leyland_county_ward, @south_ribble_parliamentary_constituency]
+      slug = LicenceLocationIdentifier.find_slug(areas, @artefact)
 
       assert_nil slug
     end
 
-    should "select the closest authority from geostack if district not provided" do
-      geostack = {
-        "council" => [
-          {"id" => 2, "name" => "South Ribble Borough Council", "type" => "DIS", "govuk_slug" => "south-ribble-borough-council"},
-          {"id" => 3, "name" => "Leyland South West", "type" => "CED" },
-          {"id" => 4, "name" => "South Ribble", "type" => "WMC" }
-        ]
-      }
-      slug = LicenceLocationIdentifier.find_slug(geostack, @artefact)
+    should "select the closest authority from areas if district not provided" do
```

Comment:
It took me a while to figure out what this test and the next few are for. I _think_ the intent is to test the fallback behaviour when there are no areas for any of the providing tiers. In this case the artefact specifies `"providing_tier" => %w(county unitary)` but neither of those are in `areas`, so instead `LicenceLocationIdentifier.find_slug` goes through [`LocationIdentifier.authority_types`](https://github.com/alphagov/frontend/blob/c8a03ac693e4036ffc04f254aaf82e01da0e4ecb/lib/location_identifier.rb#L4) looking in `areas` for an area of each type in turn, and returns the slug of the first one it finds. This fallback is also what happens when the artefact doesn't contain any providing tiers.

The "closest" phrasing of these tests was pretty confusing to me - it sounds like it's doing a geographical query rather than relying on the ordering of area types in an array. I know you didn't choose that word, but is there a better way of describing these contexts so that this behaviour is better documented?